### PR TITLE
[#347] Add setEACL session token checks

### LINKED ISSUE #347: The gateway hangs when a user tries to create a bucket with session token without SetEACL permission
When a user with session token with only PUT permission tries to create a bucket, the GW does the following steps: 
1. puts container successfully
2. tries to put eacl, but receives the error:
```
could not handle request: could not execute GetEACL request: extended ACL table is not set for this container
```
and returns 500 Internal Error to the user. 

Then AWS CLI retries to create the bucket again and receives `name is already taken` error because containers with that name is already created (see p. 1)

### PR DESCRIPTION
**Pull request by masterSplinter01**
Closes #347

### CODE REVIEW

**Review by alexvanin — APPROVED** [review]

File: api/layer/layer.go [hunk 1]
```diff
@@ -330,17 +330,19 @@ type (
...
-		Policy             *netmap.PlacementPolicy
-		EACL               *eacl.Table
-		SessionToken       *session.Token
-		LocationConstraint string
+		Name                   string
+		ACL                    uint32
+		Policy                 *netmap.PlacementPolicy
+		EACL                   *eacl.Table
+		SessionTokenForPut     *session.Token
+		SessionTokenForSetEACL *session.Token
```
Comment (KirillovDenis) [inline api/layer/layer.go:338]: I would drop `SessionTokenForSetEACL` param and use only `EACL` field

File: api/layer/layer.go [hunk 2]
```diff
@@ -330,17 +330,19 @@ type (
...
+		LocationConstraint     string
 	}
 	// PutBucketACLParams stores put bucket acl request parameters.
 	PutBucketACLParams struct {
-		Name string
-		EACL *eacl.Table
+		Name         string
+		EACL         *eacl.Table
+		SessionToken *session.Token
```
Comment (KirillovDenis) [inline api/layer/layer.go:345]: I would drop `SessionToken` param and use only `EACL` field

**Review by alexvanin — APPROVED** [review]

File: api/handler/multipart_upload.go [hunk 3]
```diff
@@ -398,31 +392,48 @@ func (h *handler) CompleteMultipartUploadHandler(w http.ResponseWriter, r *http.
...
-			}
-			if err = h.updateBucketACL(r, astObject, bktInfo); err != nil {
-				h.logAndSendError(w, "could not update bucket acl while completing multipart upload", reqInfo, err, additional...)
-				return
-			}
+	c := &layer.CompleteMultipartParams{
+		Info:  uploadInfo,
+		Parts: reqBody.Parts,
+	}
+	objInfo, err := h.obj.CompleteMultipartUpload(r.Context(), c)
```
Comment (KirillovDenis) [inline api/handler/multipart_upload.go:407]: Why did we move this invocation?

File: api/handler/util.go [hunk 4]
```diff
@@ -80,3 +81,17 @@ func parseRange(s string) (*layer.RangeParams, error) {
...
 	}, nil
 }
+
+func getSessionTokenSetEACL(ctx context.Context) (*session.Token, error) {
+	var sessionToken *session.Token
+	boxData, err := layer.GetBoxData(ctx)
+	if err != nil {
+		return nil, err
+	}
+	sessionToken = boxData.Gate.SessionTokenForSetEACL()
```
Comment (KirillovDenis) [inline api/handler/util.go:91]:
```suggestion
	boxData, err := layer.GetBoxData(ctx)
	if err != nil {
		return nil, err
	}
	sessionToken := boxData.Gate.SessionTokenForSetEACL()
```

File: api/handler/multipart_upload.go [hunk 5]
```diff
@@ -398,31 +392,48 @@ func (h *handler) CompleteMultipartUploadHandler(w http.ResponseWriter, r *http.
...
+			ObjectInfo: objInfo,
+			TagSet:     uploadData.TagSet,
+		}
+		if err = h.obj.PutObjectTagging(r.Context(), t); err != nil {
+			h.logAndSendError(w, "could not put tagging file of completed multipart upload", reqInfo, err, additional...)
+			return
+		}
+	}
+
+	if uploadData.ACL != nil {
```
[the remark this answers is not in the document]
Comment (masterSplinter01) [reply api/handler/multipart_upload.go:423]: Lost while rebase, removed `uploadData` from line 358. Now it works.

File: api/handler/multipart_upload.go (hunk 3 above)
Comment (masterSplinter01) [reply api/handler/multipart_upload.go:406]: Before we used to make the following steps:
1. Get init part info
2. Complete multipart upload
3. If init part has payload, get init part, unmarshal payload, put tagging and/or `ACL` file

If a user don't have a session token, step 3 fails, but an object is created. Thus, now we the following steps: 
1. Get init part info
2. If init part has payload and the payload contains `ACL`, check presence of `EACL` token.
3. Complete multipart upload
4. Put tagging and/or `ACL` file

**Pull request merged**